File: Java/1979_Find_Greatest_Common_Divisor_of_Array.java
class Solution {
    public int findGCD(int[] nums) {
        int max = Integer.MIN_VALUE;
        int min = Integer.MAX_VALUE;
        for(int num : nums){
            max = Math.max(max, num);
            min = Math.min(min, num);
        }
        
        return getGDC(max, min);
        /*
        for(int i = min; i > 0; i--){
            if(max%i == 0 && min%i == 0){
                return i;
            }
        }
        return 1;
        */
    }
    private int getGDC(int a, int b){
        if(b == 0){
            return a;
        }
        else{
            return getGDC(b, a%b);
        }
    }
}

File: CS/1979_Find_Greatest_Common_Divisor_of_Array.cs
public class Solution {
    public int FindGCD(int[] nums) {
        int result = 1;
        int max = Int32.MinValue;
        int min = Int32.MaxValue;
        
        foreach(var num in nums){
            max = Math.Max(max, num);
            min = Math.Min(min, num);
        }
        
        for(int i = min; i>0; i--){
            if(max%i == 0 && min%i == 0){
                result = i;
                break;
            }
        }
        return result;;
    }
}
